test/jb_projects/challenge_handle.test.js

```diff
@@ -0,0 +1,92 @@
+import { expect } from 'chai';
+import { ethers } from 'hardhat';
+import { getTimestamp } from '../helpers/utils';
+
+import { deployMockContract } from '@ethereum-waffle/mock-contract';
+
+import jbOperatoreStore from '../../artifacts/contracts/JBOperatorStore.sol/JBOperatorStore.json';
+
+describe('JBProjects::challengeHandle(...)', function () {
+  const PROJECT_HANDLE_1 = 'PROJECT_1';
+  const METADATA_CID = '';
+  const PROJECT_ID = 1;
+
+  async function setup() {
+    let [deployer, projectOwner, ...addrs] = await ethers.getSigners();
+
+    let mockJbOperatorStore = await deployMockContract(deployer, jbOperatoreStore.abi);
+
+    let jbProjectsFactory = await ethers.getContractFactory('JBProjects');
+    let jbProjectsStore = await jbProjectsFactory.deploy(mockJbOperatorStore.address);
+
+    return {
+      projectOwner,
+      deployer,
+      addrs,
+      jbProjectsStore,
+    };
+  }
+
+  it(`Should challenge handle successfully`, async function () {
+    const { projectOwner, deployer, addrs, jbProjectsStore } = await setup();
+
+    await jbProjectsStore
+      .connect(deployer)
+      .createFor(
+        /*owner=*/ projectOwner.address,
+        /*handle=*/ ethers.utils.formatBytes32String(PROJECT_HANDLE_1),
+        /*metadataCid=*/ '',
+      );
+
+    let tx = await jbProjectsStore
+      .connect(addrs[0])
+      .challengeHandle(/*handle=*/ ethers.utils.formatBytes32String(PROJECT_HANDLE_1));
+
+    let expectedChallengeExpiry = (await getTimestamp(tx.blockNumber)).add(31536000);
+    let storedChallengeExpiryOf = await jbProjectsStore
+      .connect(addrs[0])
+      .challengeExpiryOf(ethers.utils.formatBytes32String(PROJECT_HANDLE_1));
+    await expect(storedChallengeExpiryOf).equal(expectedChallengeExpiry);
+
+    await expect(tx)
+      .to.emit(jbProjectsStore, 'ChallengeHandle')
+      .withArgs(
+        ethers.utils.formatBytes32String(PROJECT_HANDLE_1),
+        PROJECT_ID,
+        expectedChallengeExpiry,
+        addrs[0].address,
```

Comment:
please be consistent with param comments. first and fourth params would need comments here.